contract/src/Custody.sol

```diff
@@ -310,54 +293,64 @@ contract Custody is IChannel, IDeposit {
     function challenge(bytes32 channelId, State calldata candidate, State[] calldata proofs) external {
         Metadata storage meta = _channels[channelId];
 
-        // Verify channel exists and is in a valid state for challenge
+        // checks
         if (meta.stage == ChannelStatus.VOID) revert ChannelNotFound(channelId);
-        if (meta.stage == ChannelStatus.FINAL) revert InvalidStatus();
-
-        // Verify that at least one participant signed the state
-        if (candidate.sigs.length == 0) revert InvalidStateSignatures();
+        if (meta.stage == ChannelStatus.DISPUTE || meta.stage == ChannelStatus.FINAL) revert InvalidStatus();
+        if (candidate.intent == StateIntent.FINALIZE) revert InvalidState();
 
-        // Validate version based on channel status
-        if (meta.stage == ChannelStatus.INITIAL && candidate.version != 0) revert InvalidState();
+        StateIntent lastValidStateIntent = meta.lastValidState.intent;
 
-        if (candidate.data.length != 0) {
-            if (candidate.intent == StateIntent.INITIALIZE) {
-                // TODO:
-            } else if (candidate.intent == StateIntent.RESIZE) {
-                uint256 deposited = meta.expectedDeposits[CLIENT_IDX].amount + meta.expectedDeposits[SERVER_IDX].amount;
-                uint256 expected = candidate.allocations[CLIENT_IDX].amount + candidate.allocations[SERVER_IDX].amount;
-                if (deposited != expected) {
-                    revert InvalidAllocations();
-                }
+        if (meta.stage == ChannelStatus.INITIAL) {
+            // main goal: verify Candidate == LastValidState, close channel
+            if (!Utils.statesAreEqual(candidate, meta.lastValidState)) {
+                revert InvalidState();
             }
-        }
 
-        if (
-            candidate.data.length == 0
-                || (candidate.intent != StateIntent.INITIALIZE && candidate.intent != StateIntent.RESIZE)
-        ) {
-            // if no state data or intent is not INITIALIZE or RESIZE, assume this is a normal state
+            _closeEffectsAndInteractions(channelId, candidate.allocations);
 
-            // Verify the state is valid according to the adjudicator
-            if (!IAdjudicator(meta.chan.adjudicator).adjudicate(meta.chan, candidate, proofs)) revert InvalidState();
+            emit Challenged(channelId, block.timestamp);
+            emit Closed(channelId, candidate);
+            return;
+        }
```

Comment:
Nothing to change here, but this part feels a little bit odd. The only way to "close" the channel, that wasn't joint by anyone is to issue challenge. And then broadcast Challenged and Closed events all together

That's just.. odd, and feels like something conceptual is missing here

Reply:
`challenge(...)` usage is forced here as it has (will have in future update) a `challengerSig` param that requires "the approval" of the challenge from one of the participants.
If we are to implement such feature with `close(...)`, then anyone would be able to close any channel before clearnode joins.

Moreover, `challenge(...)` is conceptual for single-party-initiated close, which is the case in such situation.

However, I agree with you on the fact that consecutive `Challenged(...)` and `Closed(...)` seem a little odd, nevertheless we are able to tweak this behaviour in the future.

Reply:
I agree that your suggested approach is more flexible for channels with more than 2 participants.